onchain/src/quests/rainbow_quest.cairo

```diff
@@ -0,0 +1,81 @@
+#[starknet::contract]
+pub mod RainbowQuest {
+    use starknet::{ContractAddress, get_caller_address};
+    use art_peace::{IArtPeaceDispatcher, IArtPeaceDispatcherTrait};
+    use art_peace::quests::{IQuest, IRainbowQuest, QuestClaimed};
+
+    #[storage]
+    struct Storage {
+        art_peace: ContractAddress,
+        reward: u32,
+        claimed: LegacyMap<ContractAddress, bool>,
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        QuestClaimed: QuestClaimed,
+    }
+
+    #[derive(Drop, Serde)]
+    pub struct RainbowQuestInitParams {
+        pub art_peace: ContractAddress,
+        pub reward: u32,
+    }
+
+    #[constructor]
+    fn constructor(ref self: ContractState, init_params: RainbowQuestInitParams) {
+        self.art_peace.write(init_params.art_peace);
+        self.reward.write(init_params.reward);
+    }
+
+    #[abi(embed_v0)]
+    impl RainbowQuestImpl of IRainbowQuest<ContractState> {
+        fn is_claimed(self: @ContractState, user: ContractAddress) -> bool {
+            self.claimed.read(user)
+        }
+    }
+
+    #[abi(embed_v0)]
+    impl RainbowQuest of IQuest<ContractState> {
+        fn get_reward(self: @ContractState) -> u32 {
+            self.reward.read()
+        }
+
+        fn is_claimable(
+            self: @ContractState, user: ContractAddress, calldata: Span<felt252>
+        ) -> bool {
+            if self.claimed.read(user) {
+                return false;
+            }
+
+            let art_piece = IArtPeaceDispatcher { contract_address: self.art_peace.read() };
+
+            let mut result = true;
+            let mut i = 0;
+            while i < art_piece
+                .get_color_count() {
+                    if (art_piece.get_user_pixels_placed_color(user, i) == 0) {
+                        result = false;
```

Comment:
Could you make the while loop break early when this happens, since the result will always be false?

Reply:
oh yeah, sure. My bad. 
I first tried to break with a return value `break false;` but its not available in while loop, only in for loop. But i can just break indeed, thanks!

Reply:
Yeah, I wish we could break with return values in while loops. I'm not sure why that's not a thing currently.